src/aurora/actionscript/asbuffer.cpp

```diff
@@ -0,0 +1,759 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with xoreos. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+/** @file
+ *  Buffer for handling actionscript byte code.
+ */
+
+#include <boost/none_t.hpp>
+#include <src/common/bitstream.h>
+#include <src/common/memreadstream.h>
+
+#include "src/aurora/actionscript/variable.h"
+#include "src/aurora/actionscript/asbuffer.h"
+#include "src/aurora/actionscript/function.h"
+
+namespace Aurora {
+
+namespace ActionScript {
+
+const static byte kActionNextFrame = 0x04;
+const static byte kActionPreviousFrame = 0x05;
+const static byte kActionStop = 0x07;
+const static byte kActionToggleQuality = 0x08;
+const static byte kActionStopSounds = 0x09;
+const static byte kActionSubtract = 0x0B;
+const static byte kActionMultiply = 0x0C;
+const static byte kActionDivide = 0x0D;
+const static byte kActionEquals = 0x0E;
+const static byte kActionAnd = 0x10;
+const static byte kActionOr = 0x11;
+const static byte kActionNot = 0x12;
+const static byte kActionStringEquals = 0x13;
+const static byte kActionPop = 0x17;
+const static byte kActionGetVariable = 0x1C;
+const static byte kActionSetVariable = 0x1D;
+const static byte kActionTrace = 0x26;
+const static byte kActionDefineLocal = 0x3C;
+const static byte kActionCallFunction = 0x3D;
+const static byte kActionReturn = 0x3E;
+const static byte kActionNewObject = 0x40;
+const static byte kActionInitArray = 0x42;
+const static byte kActionAdd2 = 0x47;
+const static byte kActionLess2 = 0x48;
+const static byte kActionEquals2 = 0x49;
+const static byte kActionToNumber2 = 0x4A;
+const static byte kActionPushDuplicate = 0x4C;
+const static byte kActionGetMember = 0x4E;
+const static byte kActionSetMember = 0x4F;
+const static byte kActionIncrement = 0x50;
+const static byte kActionCallMethod = 0x52;
+const static byte kActionEnumerate2 = 0x55;
+const static byte kActionStrictEquals = 0x66;
+const static byte kActionExtends = 0x69;
+const static byte kActionStoreRegister = 0x87;
+const static byte kActionConstantPool = 0x88;
+const static byte kActionDefineFunction2 = 0x8E;
+const static byte kActionPush = 0x96;
+const static byte kActionJump = 0x99;
+const static byte kActionGetURL2 = 0x9A;
+const static byte kActionDefineFunction = 0x9B;
+const static byte kActionIf = 0x9D;
+
+ASBuffer::ASBuffer(Common::SeekableReadStream *as, bool printAssembly) :
+	_seeked(0), _printAssembly(printAssembly), _script(as) {
+	assert(as);
+}
+
+void ASBuffer::run(AVM &avm) {
+	_script->seek(0);
+	execute(avm);
+}
+
+void ASBuffer::setConstantPool(std::vector<Common::UString> constantPool) {
+	_constants = constantPool;
+}
+
+void ASBuffer::execute(AVM &avm) {
+	byte opcode;
+	if (_printAssembly)
+		info("-----");
```

Comment:
That should rather be a debug channel.

See, for example, how `src/aurora/nwscript/ncsfile.cpp` handles it with `debugC()`. You should add a relevant debug channel into the `Common::DebugManager`.

(Possible, the "kDebugScripts" channel should be renamed to say that it's about NWScript. Lua (and now ActionScript) should be their own debug channels)